src/services/assets.js

```diff
@@ -62,21 +64,31 @@ function contractHasMethod(contractCode, encodedMethodName) {
   return contractCode.includes(encodedMethodName);
 }
 
-export function transferERC20(options: ERC20TransferOptions) {
+export async function transferERC20(options: ERC20TransferOptions) {
   const {
     contractAddress,
     to,
     amount,
     wallet,
-    decimals = 18,
+    decimals: defaultDecimals = 18,
     nonce,
+    gasLimit,
+    gasPrice,
   } = options;
   wallet.provider = getEthereumProvider(NETWORK_PROVIDER);
   const contract = new Contract(contractAddress, ERC20_CONTRACT_ABI, wallet);
-  if (decimals > 0) {
-    return contract.transfer(to, utils.parseUnits(amount.toString(), decimals), { nonce });
-  }
-  return contract.transfer(to, utils.bigNumberify(amount.toString()), { nonce });
+  const decimals = defaultDecimals > 0
```

Comment:
for naming purposes, this should be parsedAmount right ?

Reply:
Thanks, changed it to `contractAmount`.